src/reducers/addons/addons.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Addons reducer.
+ * @module reducers/addons/addons
+ */
+
+import {
+  INSTALL_ADDON,
+  LIST_ADDONS,
+  UNINSTALL_ADDON,
+} from '../../constants/ActionTypes';
+
+const initialState = {
+  error: null,
+  installedAddons: [],
+  availableAddons: [],
+  upgradableAddons: [],
+  loaded: false,
+  loading: false,
+};
+
+/**
+ * Case-insensitive sorting method for Addons names.
+ * @function addonsSorter
+ * @param {String} a Add-on name.
+ * @param {String} b Add-on name.
+ * @returns {Number} Comparison result.
+ */
+function addonsSorter(a, b) {
+  const titleA = a.title.toUpperCase();
+  const titleB = b.title.toUpperCase();
+  if (titleA < titleB) {
+    return -1;
+  } else if (titleA > titleB) {
+    return 1;
+  }
+  return 0;
+}
+
+/**
+ * Addons reducer.
+ * @function addons
+ * @param {Object} state Current state.
+ * @param {Object} action Action to be handled.
+ * @returns {Object} New state.
+ */
+export default function addons(state = initialState, action = {}) {
+  switch (action.type) {
+    case `${INSTALL_ADDON}_PENDING`:
+    case `${LIST_ADDONS}_PENDING`:
+    case `${UNINSTALL_ADDON}_PENDING`:
+      return {
+        ...state,
+        error: null,
+        loaded: false,
+        loading: true,
+      };
+    case `${INSTALL_ADDON}_SUCCESS`:
```

Comment:
Does the `INSTALL_ADDON` also return the full updated list? If not then you need to split the `SUCCESS` cases for these 3 actions.